stl/inc/chrono

```diff
@@ -5171,6 +5176,180 @@ namespace chrono {
+                _THROW(format_error("Invalid type specification."));
+            }
+
+            _Chrono_specs<_CharT> _Conv_spec{._Modifier = _Char_mod, ._Type = _Char_type};
+            _Specs._Chrono_specs_list.push_back(_Conv_spec);
+        }
+
+        constexpr void _On_lit_char(_CharT _Lit_ch) {
+            _Chrono_specs<_CharT> _Lit_char_spec{._Lit_char = _Lit_ch};
+            _Specs._Chrono_specs_list.push_back(_Lit_char_spec);
+        }
+
+    protected:
+        _Chrono_format_specs<_CharT>& _Specs;
+    };
+
+    template <class _CharT, _Chrono_parse_spec_callbacks<_CharT> _Callbacks_type>
+    _NODISCARD constexpr const _CharT* _Parse_conversion_specs(
+        const _CharT* _Begin, const _CharT* _End, _Callbacks_type&& _Callbacks) {
+        ++_Begin; // move past %
+        if (_Begin == _End || *_Begin == '}') {
+            _THROW(format_error("Invalid format string."));
+        }
+
+        _CharT _Mod = '\0';
+        _CharT _Ch  = *_Begin;
+
+        if (_Ch == 'E' || _Ch == 'O') { // includes modifier
+            _Mod = _Ch;
+            ++_Begin;
+            if (_Begin == _End || *_Begin == '}') {
+                _THROW(format_error("Invalid format string - missing type after modifier."));
+            }
+        }
+
+        _CharT _Type = *_Begin;
+        _Callbacks._On_conversion_spec(_Mod, _Type);
+
+        return ++_Begin;
+    }
+
+    template <class _CharT, _Chrono_parse_spec_callbacks<_CharT> _Callbacks_type>
+    _NODISCARD constexpr const _CharT* _Parse_chrono_format_specs(
+        const _CharT* _Begin, const _CharT* _End, _Callbacks_type&& _Callbacks) {
+        if (_Begin == _End || *_Begin == '}') {
+            return _Begin;
+        }
+
+        _Begin = _Parse_align(_Begin, _End, _Callbacks);
+        if (_Begin == _End) {
+            return _Begin;
+        }
+
+        _Begin = _Parse_width(_Begin, _End, _Callbacks);
+        if (_Begin == _End) {
+            return _Begin;
+        }
+
+        if (*_Begin == '.') {
+            _Begin = _Parse_precision(_Begin, _End, _Callbacks);
+            if (_Begin == _End) {
+                return _Begin;
+            }
+        }
+
+        // chrono-spec
+        while (_Begin != _End && *_Begin != '}') {
+            if (*_Begin == '%') { // conversion-spec
+                if (_Begin + 1 == _End) {
+                    _THROW(format_error("Invalid format string- missing type after %"));
+                }
+
+                _CharT _Next_ch = *(_Begin + 1);
+                switch (_Next_ch) {
+                case 'n':
+                    _Callbacks._On_lit_char('\n');
+                    ++_Begin;
+                    break;
+                case 't':
+                    _Callbacks._On_lit_char('\t');
+                    ++_Begin;
+                    break;
+                case '%':
+                    _Callbacks._On_lit_char('%');
+                    ++_Begin;
+                    break;

```

Comment:
Unless I'm confused, I believe that these cases are incorrect. Here, `*_Begin == '%'` and we haven't consumed it, we're just looking ahead at `_Next_ch`. So, the `++_Begin;` will consume the `'%'` and leave `_Begin` pointing at the `'n'`, `'t'`, or (second) `'%'`. This will cause `%n` to be parsed as literal `'\n'` followed by literal `'n'`.

In contrast, the default case of `_Begin = _Parse_conversion_specs(_Begin, _End, _Callbacks);` fully consumes `"%d"` correctly, advancing `_Begin` past both the `'%'` and `'d'`, so that `'d'` isn't parsed during the next iteration.

I believe that `_Begin += 2;` would be the minimal patch. (We can't outright consume the `'%'` and avoid the `_Next_ch` lookahead without refactoring `_Parse_conversion_specs` which expects to be initially looking at the `'%'`.)